packages/aws-cdk/lib/api/hotswap/cloudwatch-logs.ts

```diff
@@ -0,0 +1,35 @@
+import { ListStackResources } from './common';
+import { EvaluateCloudFormationTemplate } from './evaluate-cloudformation-template';
+import { CloudWatchLogEventMonitor } from './monitor/logs-monitor';
+
+export async function findCloudWatchLogGroups(
+  listStackResources: ListStackResources,
+  evaluateCfnTemplate: EvaluateCloudFormationTemplate,
+  hotswapLogMonitor: CloudWatchLogEventMonitor,
+): Promise<void> {
+
+  const resources = await listStackResources.listStackResources();
+  const allGroups = resources.filter(resource => resource.ResourceType === 'AWS::Logs::LogGroup');
+  const functions = resources.filter(resource => resource.ResourceType === 'AWS::Lambda::Function');
+  const functionLogGroups = await Promise.all(functions.map(async f => {
+    const name = await evaluateCfnTemplate.findPhysicalNameFor(f.LogicalResourceId);
+    return `/aws/lambda/${name}`;
+  }));
+
+  hotswapLogMonitor.addLogGroups(functionLogGroups);
+
+  const relevantGroups = allGroups.filter(group => {
+    const resourcesReferencingLogGroup = evaluateCfnTemplate.findReferencesTo(group.LogicalResourceId);
+    if (resourcesReferencingLogGroup.some(r => (r.Type === 'AWS::ECS::TaskDefinition' || r.Type === 'AWS::StepFunctions::StateMachine'))) {
```

Comment:
Does it really make sense to do this filtering for interesting Log Groups? What if we just observed all of them that we find in the Stack?

Reply:
Yeah I was considering this too. I think we do need a way to limit what log groups we monitor. For example, we could be creating a log group for VPC Flow Logs in the same stack, but we wouldn't want to monitor that log group. Maybe we can expand it to include other resources.

Reply:
Then we should filter out the Log Groups we want to ignore, as opposed to only observing Log Groups that are "interesting", I think.